src/core_modules/capture-core/utils/cachedDataHooks/useValidatedIDsFromCache.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import { useCallback, useEffect, useState } from 'react';
+import {
+    containsKeyInStorageAsync,
+    getCachedSingleResourceFromKeyAsync,
+} from '../../MetaDataStoreUtils/MetaDataStoreUtils';
+import { userStores } from '../../storageControllers/stores';
+
+type Props = {|
+    programId: string,
+    orgUnitId?: string,
+|}
+
+const IdTypes = Object.freeze({
+    PROGRAM: 'programId',
+    ORG_UNIT: 'orgUnitId',
+});
+
+export const useValidatedIDsFromCache = ({ programId, orgUnitId }: Props) => {
+    const [valid, setValid] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState();
+
+    const getPromises = useCallback(() => {
+        const promises = [];
+        const keys = [];
+        if (programId) {
+            keys.push({ id: programId, type: IdTypes.PROGRAM });
+            promises.push(containsKeyInStorageAsync(userStores.PROGRAMS, programId));
+        }
+        if (orgUnitId) {
+            keys.push({ id: orgUnitId, type: IdTypes.ORG_UNIT });
+            promises.push(getCachedSingleResourceFromKeyAsync(userStores.ORGANISATION_UNITS_BY_PROGRAM, programId));
+        }
+
+        Promise.all(promises)
+            .then((adapterResponses) => {
+                if (orgUnitId) {
```

Comment:
Is `orgUnit` response always be inside `adapterResponses[1]`. How about the case when `programId` is `null` or `undefined` and there is only one promise to fulfill in the array? 🤔

Reply:
This should not be a problem. The orgUnit-search is in a program-context, so it should fail if a programId is missing or not valid. Either way, this has been changed, so it should not be a worry.